server/game/cards/agendas/thepowerofwealth.js

```diff
@@ -0,0 +1,48 @@
+const _ = require('underscore');
+
+const AgendaCard = require('../../agendacard.js');
+
+class ThePowerOfWealth extends AgendaCard {
+    constructor(owner, cardData) {
+        super(owner, cardData);
+
+        this.registerEvents(['onDecksPrepared']);
+    }
+
+    setupCardAbilities(ability) {
+        this.persistentEffect({
+            targetType: 'player',
+            targetController: 'current',
+            effect: ability.effects.reduceFirstMarshalledOrPlayedCardCostEachRound(1, card => card.isFaction(this.controller.getFaction()))
+        });
+    }
+
+    onDecksPrepared() {
+        let factionsInDecks = [];
+
+        this.controller.allCards.each(card => {
+            if(!factionsInDecks.includes(card.getPrintedFaction())) {
+                factionsInDecks.push(card.getPrintedFaction());
+            }
+        });
+
+        let factionToAnnounce = _.filter(factionsInDecks, faction => faction !== this.controller.getFaction() && faction !== 'neutral');
+
+        if(factionToAnnounce.length > 1) {
+            this.game.addMessage('{0} has included more than the maximum allowed number of factions for {1} in their draw and/or plot deck',
```

Comment:
While I think such an error message is fine, it's probably better if you also announce which factions they did include. The deck the player is playing with will already be marked as invalid from the validation rules of Power for Wealth.